File: wagerproof-ios-native/Wagerproof/Features/Agents/Creation/Inputs/ArchetypeCard.swift
import SwiftUI
import WagerproofDesign
import WagerproofModels
import WagerproofServices

/// Tappable preset archetype tile shown in Screen 1's "Use a Preset" path.
/// Ports `components/agents/inputs/ArchetypeCard.tsx`.
struct ArchetypeCard: View {
    /// `.standard` — the wizard tile: emoji glyph + recommended-sport badges.
    /// `.compact` — the onboarding tile: a pixel-office character in place of
    /// the emoji and no sport badges, so the row reads slimmer.
    enum Style { case standard, compact }

    let row: PresetArchetypeRow
    var style: Style = .standard
    /// Pixel-office character shown in `.compact` style. Ignored by `.standard`,
    /// which renders `row.emoji`.
    var spriteIndex: Int = 0
    let selected: Bool
    let onSelect: () -> Void

    private var accentColor: Color {
        Color(hexString: row.color) ?? Color(hex: 0x00E676)
    }

    private var isCompact: Bool { style == .compact }

    var body: some View {
        Button(action: onSelect) {
            VStack(alignment: .leading, spacing: 12) {
                HStack(alignment: .top, spacing: 12) {
                    leadingGlyph

                    VStack(alignment: .leading, spacing: 4) {
                        Text(row.name)
                            .font(.system(size: 17, weight: .bold))
                            .foregroundStyle(Color.appTextPrimary)
                            .lineLimit(1)
                        Text(row.description)
                            .font(.system(size: 13))
                            .foregroundStyle(Color.appTextSecondary)
                            .lineLimit(2)
                            .multilineTextAlignment(.leading)
                    }

                    Spacer(minLength: 8)

                    if selected {
                        ZStack {
                            Circle().fill(accentColor)
                            Image(systemName: "checkmark")
                                .font(.system(size: 12, weight: .bold))
                                .foregroundStyle(.white)
                        }
                        .frame(width: 24, height: 24)
                    }
                }

                // Onboarding tiles drop the sport badges to stay slim; the
                // wizard keeps them so the preset's leagues read at a glance.
                if !isCompact, !row.recommendedSports.isEmpty {
                    HStack(spacing: 6) {
                        ForEach(row.recommendedSports, id: \.self) { sport in
                            sportBadge(sport)
                        }
                    }
                }
            }
            .padding(16)
            .frame(maxWidth: .infinity, alignment: .leading)
            // Liquid Glass card surface; the accent stroke remains as the
            // selection indicator only.
            .liquidGlassBackground(
                in: RoundedRectangle(cornerRadius: 16, style: .continuous),
                tint: selected ? accentColor.opacity(0.16) : Color.white.opacity(0.05)
            )
            .overlay(
                RoundedRectangle(cornerRadius: 16, style: .continuous)
                    .strokeBorder(selected ? accentColor : Color.clear, lineWidth: 2)
            )
        }
        .buttonStyle(.plain)
        .padding(.vertical, 4)
    }

    /// The 48×48 Liquid Glass tile leading the row. `.standard` shows the
    /// archetype's emoji; `.compact` puts a pixel-office character inside the
    /// same glass container (it bobs when the tile is selected).
    @ViewBuilder
    private var leadingGlyph: some View {
        switch style {
        case .standard:
            Text(row.emoji)
                .font(.system(size: 24))
                .frame(width: 48, height: 48)
                .liquidGlassBackground(
                    in: RoundedRectangle(cornerRadius: 12, style: .continuous),
                    tint: accentColor.opacity(0.20)
                )
        case .compact:
            PixelSpriteAvatar(spriteIndex: spriteIndex, animated: selected)
                .padding(5)
                .frame(width: 48, height: 48)
                .liquidGlassBackground(
                    in: RoundedRectangle(cornerRadius: 12, style: .continuous),
                    tint: accentColor.opacity(0.20)
                )
        }
    }

    private func sportBadge(_ sport: AgentSport) -> some View {
        HStack(spacing: 4) {
            Image(systemName: sport.sfSymbol)
                .font(.system(size: 11))
            Text(sport.label)
                .font(.system(size: 11, weight: .bold))
        }
        .padding(.horizontal, 8)
        .padding(.vertical, 5)
        .foregroundStyle(Color.appTextSecondary)
        .background(
            RoundedRectangle(cornerRadius: 8, style: .continuous)
                .fill(Color.appBorder.opacity(0.4))
        )
        .overlay(
            RoundedRectangle(cornerRadius: 8, style: .continuous)
                .strokeBorder(Color.appBorder, lineWidth: 1)
        )
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/agents/creation/inputs/ArchetypeCard.kt
package com.wagerproof.app.features.agents.creation.inputs

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Check
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.app.features.agents.colorFromHexString
import com.wagerproof.app.features.agents.iconVector
import com.wagerproof.core.design.components.liquidGlassBackground
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.services.PresetArchetypeRow

/**
 * Tappable preset archetype tile in Step 1's "Use a Preset" path. Port of iOS
 * `ArchetypeCard`. Accent color = the row's hex color (fallback brand green).
 */
@Composable
fun ArchetypeCard(
    row: PresetArchetypeRow,
    selected: Boolean,
    onSelect: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val accentColor = colorFromHexString(row.color) ?: AppColors.brandGreenBright
    val cardShape = RoundedCornerShape(16.dp)

    Column(
        modifier = modifier
            .fillMaxWidth()
            .padding(vertical = 4.dp)
            .liquidGlassBackground(
                shape = cardShape,
                tint = if (selected) accentColor.copy(alpha = 0.16f) else Color.White.copy(alpha = 0.05f),
            )
            .border(
                if (selected) 2.dp else 0.dp,
                if (selected) accentColor else Color.Transparent,
                cardShape,
            )
            .clickableNoRipple(onSelect)
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Row(verticalAlignment = Alignment.Top) {
            Box(
                modifier = Modifier
                    .size(48.dp)
                    .liquidGlassBackground(
                        shape = RoundedCornerShape(12.dp),
                        tint = accentColor.copy(alpha = 0.20f),
                    ),
                contentAlignment = Alignment.Center,
            ) {
                Text(text = row.emoji, fontSize = 24.sp)
            }

            Spacer(Modifier.size(12.dp))

            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = row.name,
                    color = AppColors.appTextPrimary,
                    fontSize = 17.sp,
                    fontWeight = FontWeight.Bold,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                )
                Text(
                    text = row.description,
                    color = AppColors.appTextSecondary,
                    fontSize = 13.sp,
                    maxLines = 2,
                    overflow = TextOverflow.Ellipsis,
                )
            }

            if (selected) {
                Spacer(Modifier.size(8.dp))
                Box(
                    modifier = Modifier
                        .size(24.dp)
                        .clip(RoundedCornerShape(50))
                        .background(accentColor),
                    contentAlignment = Alignment.Center,
                ) {
                    Icon(
                        imageVector = Icons.Filled.Check,
                        contentDescription = null,
                        tint = Color.White,
                        modifier = Modifier.size(14.dp),
                    )
                }
            }
        }

        if (row.recommendedSports.isNotEmpty()) {
            Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
                row.recommendedSports.forEach { sport ->
                    Row(
                        modifier = Modifier
                            .background(AppColors.appBorder.copy(alpha = 0.4f), RoundedCornerShape(8.dp))
                            .border(1.dp, AppColors.appBorder, RoundedCornerShape(8.dp))
                            .padding(horizontal = 8.dp, vertical = 5.dp),
                        horizontalArrangement = Arrangement.spacedBy(4.dp),
                        verticalAlignment = Alignment.CenterVertically,
                    ) {
                        Icon(
                            imageVector = sport.iconVector(),
                            contentDescription = null,
                            tint = AppColors.appTextSecondary,
                            modifier = Modifier.size(11.dp),
                        )
                        Text(
                            text = sport.label,
                            color = AppColors.appTextSecondary,
                            fontSize = 11.sp,
                            fontWeight = FontWeight.Bold,
                        )
                    }
                }
            }
        }
    }
}
